lib/main.js

```diff
@@ -38,12 +37,32 @@ class TypeScriptLanguageClient extends AutoLanguageClient {
 
   shouldStartForEditor(editor) {
     if (atom.config.get('ide-typescript.ignoreFlow') === true) {
-      const flowConfigPath = path.join(this.getProjectPath(editor.getURI() || ''), '.flowconfig');
-      if (fs.existsSync(flowConfigPath)) return false;
+      const flowConfigPath = path.join(this.getProjectPath(editor.getURI() || ''), '.flowconfig')
+      if (fs.existsSync(flowConfigPath)) return false
     }
+
+    if (!this.validateTypeScriptServerPath()) return false
+
     return super.shouldStartForEditor(editor);
   }
 
+  validateTypeScriptServerPath() {
+    const tsServerPath = atom.config.get('ide-typescript.typeScriptServer.path')
+    if (fs.existsSync(tsServerPath)) return true
```

Comment:
Hi! I'm not involved here or familiar with this code at all (sorry to just drop in like this), but I was playing around with this branch and noticed that on my machine this doesn't work because `process.cwd()` here is `/`. No matter what, I get the warning. It looks like this needs a `path.join` like `shouldStartForEditor` above.

But other than that this is working great for me! Thanks for your hard work!

Reply:
No problem just dropping in :)  I'll get that fixed before merging - it interestingly works here simply because I'm working on ide-typescript and so had launched atom from that folder :D

Most of the work was by @mattlyons0 so my hat is tipped to him.